### ISSUE TITLE
false negatives in the Java rule weak-random

### ISSUE THREAD

**Issue by 9iang22** [returntocorp/sgrep-rules#3732]
**Describe the bug**
There are some false negatives in the Java rule [weak-random](https://semgrep.dev/playground/r/java.lang.security.audit.crypto.weak-random.weak-random).

**To Reproduce**

The following code should be matched as weak-random, but is currently not detected:
```
private static final java.util.Random rng = new java.util.Random();

public static int Random() {		
	return (Math.abs(rng.nextInt())) & 0xFFFF;
}

public static long Random32() {		
	return Math.abs(rng.nextLong()) & 0xFFFFFFFFL;
}
```

**Priority**
How important is this to you?
- [ ] P0: blocking me from making progress
- [x] P1: this will block me in the near future
- [ ] P2: annoying but not blocking me

**Additional Context**

The current pattern in the rule is:
```yaml
pattern-either:
  - pattern: new java.util.Random(...).$FUNC(...) 
  - pattern: java.lang.Math.random(...)
 ```
This misses cases where java.util.Random is instantiated but not directly chained with a method call in the same expression.

It can be refined simply:
```yaml
pattern-either:
  - pattern: new java.util.Random(...)
  - pattern: java.lang.Math.random(...)
```
This change will match `java.util.Random` instantiations, covering the examples above and aligning better with the rule's description.

I’m happy to submit a PR with this update if the approach is approved.
